src/context/WalletProvider/KeepKey/components/Label.tsx

```diff
@@ -4,8 +4,16 @@ import { RecoverDevice, ResetDevice } from '@shapeshiftoss/hdwallet-core'
 import { useRef, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 import { Text } from 'components/Text'
+import { VALID_ENTROPY } from 'context/WalletProvider/KeepKey/components/RecoverySettings'
 import { useWallet } from 'hooks/useWallet/useWallet'
 
+const parseIntToEntropy = (entropy: string): RecoverDevice['entropy'] => {
+  const isValidEntropyNumber = (entropy: number): entropy is RecoverDevice['entropy'] =>
+    new Set(VALID_ENTROPY.map(parseInt)).has(entropy)
+  const parsedInt = parseInt(entropy)
+  return isValidEntropyNumber(parsedInt) ? parsedInt : 128
+}
```

Comment:
Why is this so complicated? Checking whether the value is one of 3 possibilities shouldn't require so much code.

Reply:
it's checking and type casting in one to a union of set values

Reply:
like I said, why is it so complicated?

Reply:
I hear that - it's done this way to give us a single source of truth in `web` for what a valid entropy is (`VALID_ENTROPY_NUMBERS` in `RecoverySettings.tsx`), rather than having magic numbers throughout that need to be updated whenever what is considered a valid entropy changes.

Agree though, it makes it a little verbose and complicated - with what I've said above considered I'm open to either approach.

I've made some improvements in https://github.com/shapeshift/web/pull/1561/commits/d3474e4092cca0e2baad79bcc6fb05f1aba5d710 to make the current implementation clearer.